packages/admin-ui/src/tooltip/tooltip.tsx

```diff
@@ -0,0 +1,89 @@
+import type { FunctionComponentElement } from 'react'
+import React, { cloneElement } from 'react'
+import { useSystem } from '@vtex/admin-ui-react'
+import type { TooltipStateProps } from 'ariakit/Tooltip'
+import {
+  useTooltipState,
+  Tooltip as TooltipPopover,
+  TooltipAnchor,
+  TooltipArrow,
+} from 'ariakit'
+
+import * as style from './tooltip.style'
+import { TooltipTrigger } from './tooltip-trigger'
+
+/**
+ * Popup that displays information related to an element on :focus (by keyboard) or :hover (by mouse)
+ * @example
+ * <Tooltip label="Label" />
+ */
+export function Tooltip(props: TooltipProps) {
+  const {
+    text,
+    visible,
+    children: _children,
+    placement = 'bottom',
+    bleedX = false,
+    bleedY = false,
+  } = props
+
+  const hasChildren = !!_children
+
+  const { cn } = useSystem()
+  const state = useTooltipState({
+    placement,
+    visible,
+    gutter: hasChildren ? 2 : 0,
+  })
+
+  const children: FunctionComponentElement<unknown> = _children ?? (
+    <TooltipTrigger bleedX={bleedX} bleedY={bleedY} />
+  )
+
+  return (
+    <>
+      <TooltipAnchor
+        state={state}
+        {...(children.props as any)}
```

Comment:
Why was the any necessary?

Reply:
Could _children get some more specific type that wouldn't cause a conflict?

Reply:
Changed to `tooltipAnchorChildren`. The any is there for a typescript edge case.